docs/book/user-guide/best-practices/quick-wins.md

```diff
@@ -321,7 +341,163 @@ zenml artifact-store register my_store \
+* **Consistent environments** - Same Docker containerization as production
+* **Reduced friction** - No cloud provisioning delays or permission issues during development
+
+**Best practices**
+* Create a small representative dataset for smoke testing
+* Use configuration parameters to enable smoke-test mode
+* Keep dependencies identical between smoke tests and production
+* Run the exact same pipeline code locally and in the cloud
+* Store sample data in version control for reliable testing
+* Use `prints` or logging to clearly indicate when running in smoke-test mode
+
+This approach works best when you design your pipelines to be configurable from the start, allowing them to run with reduced data size, shorter training cycles, or simplified processing steps during development.
+
+Learn more: [Local Docker Orchestrator](https://docs.zenml.io/stack-components/orchestrators/local-docker)
+
+## 9 Organize with tags
+
+**Why** -- add flexible, searchable labels to your ML assets that bring order to chaos as your project grows. Tags provide a lightweight organizational system that helps you filter pipelines, artifacts, and models by domain, status, version, or any custom category—making it easy to find what you're looking for in seconds.
+
+```python
+from zenml import pipeline, step, add_tags, Tag
+
+# 1. Tag your pipelines with meaningful categories
+@pipeline(tags=["fraud-detection", "training", "financial"])
+def training_pipeline():
+    # Your pipeline steps
+    preprocess_step(...)
+    train_step(...)
+    evaluate_step(...)
+
+# 2. Create "exclusive" tags for state management
+@pipeline(tags=[
+    Tag(name="production", exclusive=True),  # Only one pipeline can be "production"
+    "financial"
+])
+def production_pipeline():
+    pass
+
+# 3. Tag artifacts programmatically from within steps
+@step
+def evaluate_step():
+    # Your evaluation code here
+    accuracy = 0.95
+    
+    # Tag based on performance
+    if accuracy > 0.9:
+        add_tags(tags=["high-accuracy"], infer_artifact=True)
+    
+    # Tag with metadata values
+    add_tags(tags=[f"accuracy-{int(accuracy*100)}"], infer_artifact=True)
+    
+    return accuracy
+
+# 4. Use cascade tags to apply pipeline tags to all artifacts
+@pipeline(tags=[Tag(name="experiment-12", cascade=True)])
+def experiment_pipeline():
+    # All artifacts created in this pipeline will also have the "experiment-12" tag
+    pass
+```
+
+**Key features**
+* **Filter and search** - Quickly find all assets related to a specific domain or project
+* **Exclusive tags** - Create tags where only one entity can have the tag at a time (perfect for "production" status)
+* **Cascade tags** - Apply pipeline tags automatically to all artifacts created during execution
+* **Flexible organization** - Create any tagging system that makes sense for your projects
+* **Multiple entity types** - Tag pipelines, runs, artifacts, models, and run templates
+
+**Common tag operations**
```

Comment:
could use a screenshot of filtering by tags?

Reply:
Done